cmd/rebase.go

```diff
@@ -0,0 +1,99 @@
+package cmd
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/vanilla-os/abroot/core"
+	"github.com/vanilla-os/orchid/cmdr"
+)
+
+func NewRebaseCommand() *cmdr.Command {
+	cmd := cmdr.NewCommand(
+		"rebase <name>",
+		abroot.Trans("rebase.long"),
+		abroot.Trans("rebase.short"),
+		rebase,
+	)
+
+	cmd.WithBoolFlag(
+		cmdr.NewBoolFlag(
+			"dry-run",
+			"d",
+			abroot.Trans("rebase.dryRunFlag"),
+			false,
+		))
+
+	cmd.WithBoolFlag(
+		cmdr.NewBoolFlag(
+			"delete-old-system",
+			"",
+			abroot.Trans("rebase.deleteOld"),
+			false,
+		))
+
+	cmd.Args = cobra.ExactArgs(1)
+	cmd.Example = "abroot rebase vanilla-os/desktop:main"
+
+	return cmd
+}
+
+func rebase(cmd *cobra.Command, args []string) error {
+
+	if !core.RootCheck(false) {
+		cmdr.Error.Println(abroot.Trans("rebase.rootRequired"))
+		return nil
+	}
+
+	dryRun, err := cmd.Flags().GetBool("dry-run")
+	if err != nil {
+		cmdr.Error.Println(err)
+		return err
+	}
+
+	freeSpace, err := cmd.Flags().GetBool("delete-old-system")
+	if err != nil {
+		cmdr.Error.Println(err)
+		return err
+	}
+
+	name := args[0]
+	abSys, err := core.NewABSystem()
+	if err != nil {
+		cmdr.Error.Println(err)
+		return err
+	}
+
+	pkgM, err := core.NewPackageManager(dryRun)
+	if pkgM.Status == core.PKG_MNG_ENABLED {
+		var removePackages string
+		cmdr.Info.Print(abroot.Trans("rebase.removePackages"), " (y/N) ", " ")
+		fmt.Scanln(&removePackages)
+		if strings.Contains(removePackages, "y") && !dryRun {
+			addedPackages, err := pkgM.GetAddPackages()
+			if err != nil {
+				return err
+			}
+
+			for _, v := range addedPackages {
+				pkgM.Remove(v)
+			}
+			abSys.RunOperation(core.APPLY, freeSpace)
```

Comment:
Running an upgrade will apply all packages automatically, so an apply operation is not necessary here.